include/deal.II/differentiation/sd/symengine_scalar_operations.h

```diff
@@ -146,8 +146,317 @@ namespace Differentiation
+     * corresponding @p values. The modified symbol will have the key given by
+     * the first element of  @p symbol_value and the value given by its second
+     * element.
+     *
+     * This function may be used to safely transform an existing or null
+     * symbolic map (one with uninitialized entries) into one that can be used
+     * to conduct symbolic substitution operations (i.e., a substitution map).
+     */
+    template <typename SymbolicType = SD::Expression, typename ValueType>
+    void
+    set_value_in_symbol_map(
+      types::substitution_map &                 substitution_map,
+      const std::pair<SymbolicType, ValueType> &symbol_value);
+
+    /**
+     * Find the input @p symbols in the @p substitution_map and set their
+     * corresponding @p values, followed by the same operation for the
+     * @p other_symbol_values.
+     *
+     * This function may be used to safely transform an existing or null
+     * symbolic map (one with uninitialized entries) into one that can be used
+     * to conduct symbolic substitution operations (i.e., a substitution map).
+     */
+    template <typename SymbolicType = SD::Expression,
+              typename ValueType,
+              typename... Args>
+    void
+    set_value_in_symbol_map(
+      types::substitution_map &                 substitution_map,
+      const std::pair<SymbolicType, ValueType> &symbol_value,
+      const Args &... other_symbol_values);
+
+    /**
+     * Find the input @p symbols in the @p substitution_map and set their
+     * corresponding @p values. The modified symbol will have the key given by
+     * the first element of each paired entry in the @p symbol_values vector
+     * and the value given by its respective second element.
+     *
+     * This function may be used to safely transform an existing or null
+     * symbolic map (one with uninitialized entries) into one that can be used
+     * to conduct symbolic substitution operations (i.e., a substitution map).
+     */
+    template <typename SymbolicType = SD::Expression, typename ValueType>
+    void
+    set_value_in_symbol_map(
+      types::substitution_map &                              substitution_map,
+      const std::vector<std::pair<SymbolicType, ValueType>> &symbol_values);
+
+    /**
+     * Find the input @p symbols in the @p substitution_map and set their
+     * corresponding @p values. The modified symbol will have the key given by
+     * the each element the @p symbol_values map and the value given by its
+     * respective mapped element.
+     *
+     * This function may be used to safely transform an existing or null
+     * symbolic map (one with uninitialized entries) into one that can be used
+     * to conduct symbolic substitution operations (i.e., a substitution map).
+     */
+    void
+    set_value_in_symbol_map(types::substitution_map &      substitution_map,
+                            const types::substitution_map &symbol_values);
```

Comment:
These are a *lot* of variations of the same function. Do you really need them all or could one get away with fewer of these by just calling `set_value_in_symbolic_map(map, {a,b})` instead of `set_value_in_symbolic_map(map, a,b)` and similar?

There is nothing wrong with having more variations, but it's going to be more difficult to find the one you want in the documentation if there are many.

Reply:
I have found each of these useful in different contexts so I would advocate that we keep them. I also can't quite recall right now, but I think that I might need some of these variations later on for some other classes that are still to be implemented.